Django Tutorial Part 4: Django admin site

ryhoangf opened the pull request:
Django admin site's tutorial

nhihy-0598 commented:
ready

chienpv-3590 commented on catalog/admin.py:
```diff
@@ -1,3 +1,41 @@
 
-# Register your models here.
+#admin.site.register(Book)
+#admin.site.register(Author)
```
những comment này k dùng thì xóa đi

chienpv-3590 commented on catalog/admin.py:
```diff
@@ -1,3 +1,41 @@
+    display_genre.short_description = 'Genre'
+    
+
+    
```
Xóa những dòng thừa không cần thiết

ryhoangf committed:
Django Tutorial Part 4: Django admin site

Django Tutorial Part 4: Django admin site

Django Tutorial Part 2: Creating a skeleton website

Ignore .env file

Django Tutorial Part 4: Django admin site

Django Tutorial Part 4: Django admin site

nhihy-0598 commented:
ready

nhihy-0598 commented:
ready

nhihy-0598 commented:
ready

nhihy-0598 commented:
ready

huongnt-2545 commented on .env:
```diff
@@ -0,0 +1,6 @@
+   DJANGO_SECRET_KEY="django-insecure-7dua7_h=21=$i73wu4eqk*ac7n1pa5+s(4so3nu-5t46^^sup$"
+   MYSQL_DATABASE=locallibrary
+   MYSQL_USER=root
+   MYSQL_PASSWORD=ryhoangf
+   MYSQL_HOST=localhost
+   MYSQL_PORT=3306
```
git ignore file này, tạo file clone `.env.example` để đưa lên cấu trúc biến env

huongnt-2545 commented on catalog/admin.py:
```diff
@@ -1,3 +1,36 @@
+admin.site.register(Genre)
+admin.site.register(Language)
+
+class AuthorAdmin(admin.ModelAdmin):
+    list_display = ('last_name', 'first_name', 'date_of_birth', 'date_of_death')
+    fields = ['first_name', 'last_name', ('date_of_birth', 'date_of_death')]
+
+admin.site.register(Author, AuthorAdmin)
+
+class BooksInstanceInline(admin.TabularInline):
+    model = BookInstance
+
+@admin.register(Book)
+class BookAdmin(admin.ModelAdmin):
+    list_display = ('title', 'author', 'display_genre')
+    inlines = [BooksInstanceInline]
+
+    def display_genre(self, obj):
+        return ', '.join(genre.name for genre in obj.genre.all()[:3])
+    display_genre.short_description = 'Genre'
+
+@admin.register(BookInstance)
+class BookInstanceAdmin(admin.ModelAdmin):
```
mình dùng đồng nhất 1 cách register admin site e nha, dùng decorator `@admin.register(ClassName)`

nhihy-0598 commented:
## Checklist tự review pull trước khi ready để trainer review
- [ ] Sử dụng thụt lề 2 spaces/4 spaces đồng nhất ở tất cả các files (setting lại vscode /sublime text/ ... nếu chưa cài đặt)
- [ ] Cuối mỗi file kiểm tra có end line (khi đẩy lên git xem file change không bị lỗi tròn đỏ ở cuối file)
- [ ] Mỗi dòng nếu quá dài, cần xuống dòng (maximum: 80 kí tự mỗi dòng, setting trong IDE)
- [ ] gitignore các file chứa thông tin nhạy cảm (VD: .env, ..), file .pyc (python cache), ...
- [ ] Tham khảo Python coding convention https://peps.python.org/pep-0008/
- [ ] Kiểm tra mỗi pull request chỉ 1 commit, nếu nhiều hơn 1 commit thì hãy gộp commit thành 1 rồi đẩy lại lên git
- [ ] Install [PEP8](https://pypi.org/project/pep8/) ở local để check và fix các lỗi liên quan đến syntax, coding convention. Khi gửi thì chụp ảnh PASS đính kèm trong pull
- [ ] Nếu làm việc nhóm trong project thì mỗi pull cần **ít nhất 1 APPROVED** từ thành viên trong nhóm

## Related Tickets
- ticket redmine link

## WHAT (optional)
- Change number items `completed/total` in admin page.

## HOW
- I edit js file, inject not_vary_normal items in calculate function.

## WHY (optional)
- Because in previous version - number just depends on `normal` items. But in new version, we have `state` and `confirm_state` depends on both `normal` + `not_normal` items.

## Evidence (Screenshot or Video)


## Notes (Kiến thức tìm hiểu thêm)
